src/briefcase/integrations/download.py

```diff
@@ -97,3 +87,38 @@ def file(self, url, download_path, role=None):
             raise NetworkFailure(f"download {description}") from e
 
         return filename
+
+    def _fetch_and_write_content(self, response, filename):
+        """Write the content from the Requests response to file.
+
+        The data is written in to a temporary file in a filesystem location
+        designated by the platform. This avoids partially downloaded files
+        masquerading as complete downloads in later Briefcase runs.
+        The temporary file is only moved to ``filename`` if the download
+        is successful; otherwise, it is deleted.
+
+        :param response: ``requests.Response``
+        :param filename: full filesystem path to save data
+        """
+        try:
+            with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+                total = response.headers.get("content-length")
+                if total is None:
+                    temp_file.write(response.content)
+                else:
+                    progress_bar = self.tools.input.progress_bar()
+                    task_id = progress_bar.add_task("Downloader", total=int(total))
+                    with progress_bar:
+                        for data in response.iter_content(chunk_size=1024 * 1024):
+                            temp_file.write(data)
+                            progress_bar.update(task_id, advance=len(data))
+
+            self.tools.shutil.move(temp_file.name, filename)
+            # retrieving the current umask requires changing it...
+            os.umask(current_umask := os.umask(0o022))
+            self.tools.os.chmod(filename, 0o664 & ~current_umask)
```

Comment:
The `NamedTemporaryFile` file is only created with `600` permissions. So, all this `umask` stuff restores the previous behavior from `filename.open()`.

Reply:
A longer explanatory comment would be helpful here. The interplay between the 0o022 mask and the `and not` with 0o644 wan't clear to me.

There's also no test that the resulting file mask is actually followed; there's code coverage, but no real check of the logic.

Reply:
Added (some probably overly verbose) comments about umask. I also tried to implement file permission verification in the tests.....things are really wonky on windows, though...